Source/Diagnostics/ComputeDiagFunctors/BackTransformParticleFunctor.cpp

```diff
@@ -15,29 +15,39 @@
 #include <AMReX_Print.H>
 #include <AMReX_BaseFwd.H>
 
-SelectParticles::SelectParticles (const WarpXParIter& a_pti, TmpParticles& tmp_particle_data,
-                                  amrex::Real current_z_boost, amrex::Real old_z_boost,
-                                  int a_offset)
+SelectParticles::SelectParticles (
+    const WarpXParticleContainer& /*pc*/,
+    WarpXParIter& a_pti,
+    amrex::Real current_z_boost,
+    amrex::Real old_z_boost,
+    int a_offset
+)
     : m_current_z_boost(current_z_boost), m_old_z_boost(old_z_boost)
 {
     m_get_position = GetParticlePosition<PIdx>(a_pti, a_offset);
 
-    const auto lev = a_pti.GetLevel();
-    const auto index = a_pti.GetPairIndex();
-
-    zpold = tmp_particle_data[lev][index][TmpIdx::zold].dataPtr();
+    zpold = a_pti.GetAttribs("z_n_btd").dataPtr();
 }
 
 
-LorentzTransformParticles::LorentzTransformParticles ( const WarpXParIter& a_pti,
-                                TmpParticles& tmp_particle_data,
-                                amrex::Real t_boost, amrex::Real dt,
-                                amrex::Real t_lab, int a_offset)
-    : m_t_boost(t_boost), m_dt(dt), m_t_lab(t_lab)
+LorentzTransformParticles::LorentzTransformParticles (
+    const WarpXParticleContainer& /*pc*/,
+    WarpXParIter& a_pti,
+    amrex::Real t_boost,
+    amrex::Real dt,
+    amrex::Real t_lab,
+    int a_offset
+)
+    : m_t_boost(t_boost), m_dt(dt), m_t_lab(t_lab),
+      m_gammaboost(WarpX::gamma_boost), m_betaboost(WarpX::beta_boost),
+      m_Phys_c(PhysConst::c), m_inv_c2(amrex::Real(1.0)/(m_Phys_c * m_Phys_c)),
+      m_uzfrm(-m_gammaboost*m_betaboost*m_Phys_c)
 {
     using namespace amrex::literals;
 
-    if (tmp_particle_data.empty()) { return; }
+    // FIXME
+    //if (tmp_particle_data.empty()) { return; }
+
```

Comment:
@atmyers 

How do you think we should handle this `if` condition that is currently commented out?

Reply:
I think we should replace this with an assertion that the BTD components are found in the input particle iterator.

Reply:
Actually it will trigger an assertion failure below, when it tries to access them. So I think this is fine to just remove.